File: altgraph_data_app/src/main/java/org/cjoakim/cosmos/altgraph/data/common/repository/MaintainerRepository.java
package org.cjoakim.cosmos.altgraph.data.common.repository;

import com.azure.spring.data.cosmos.repository.CosmosRepository;
import org.cjoakim.cosmos.altgraph.data.common.model.npm.Maintainer;
import org.springframework.stereotype.Component;
import org.springframework.stereotype.Repository;

/**
 * This class is a Spring Data Repository for CosmosDB NPM Maintainer documents.
 * Chris Joakim, Microsoft, November 2022
 */

@Component
@Repository
public interface MaintainerRepository extends CosmosRepository<Maintainer, String> {

}

File: dotnet/src/altgraph_shared_app/Repositories/Npm/MaintainerRepository.cs
using altgraph_shared_app.Models.Npm;
using Microsoft.Azure.CosmosRepository;

namespace altgraph_shared_app.Services.Repositories.Npm
{
  public class MaintainerRepository
  {
    public IRepository<Maintainer> Maintainers { get; private set; }
    public MaintainerRepository(IRepository<Maintainer> maintainers)
    {
      Maintainers = maintainers;
    }
  }
}